packages/machine-extractor/src/MachineExtractResult.ts

```diff
@@ -708,6 +709,46 @@ export class MachineExtractResult {
     return delays.toObject();
   };
 
+  getAllMachineImplementations = () => {
+    if (
+      !this.machineCallResult.options ||
+      t.isObjectExpression(this.machineCallResult.options)
+    ) {
+      return { actions: {}, actors: {}, guards: {}, delays: {} };
+    }
+
+    const out: Record<
+      'actions' | 'actors' | 'guards' | 'delays',
+      Record<string, string>
+    > = { actions: {}, actors: {}, guards: {}, delays: {} } as const;
+
+    const foundKeyToOutKey: Record<string, keyof typeof out> = {
```

Comment:
nit: IMHO this is a little bit overcomplicated for what it is. A much simpler version would be something like:
```ts
return {
  actors: { ...getThing(this.machineCallResult.options.services), ...getThing(this.machineCallResult.options.actors) },
  actions: getThing(this.machineCallResult.options.actions)
  delays: getThing(this.machineCallResult.options.delays),
  guards: getThing(this.machineCallResult.options.guards)
}
```

Reply:
Sure, I'll update.